doc/libpmemobj/oid_is_null.3.md

```diff
@@ -116,6 +118,21 @@ The **OID_IS_NULL**() macro checks if *PMEMoid* represents a NULL object.
 
 The **OID_EQUALS**() macro compares two *PMEMoid* objects.
 
+For special cases where volatile (transient) variables need to be stored on
+persistent memory, there's a mechanism composed of *struct pmemvlt* type and
+**pmemobj_direct_volatile()** function. To use it, the *struct pmemvlt* needs to
+be placed at the beginning of the region of transient data. The *PMEMvlt* macro
+can be used to construct such a region.
+When the **pmemobj_direct_volatile()** function is called on a *struct pmemvlt*,
+it will return the pointer to the data and it will ensure that the provided
+constructor function is called exactly once in the current instance of the
+application.
+The constructor is called with the *ptr* pointer to the data that is located
+immediately after the *struct pmemvlt*, and this function will return the same
```

Comment:
This means that offset calculation must be on the pmemobj side...

I think compiler can insert padding between run_id and the next field when its alignment is bigger than 8 bytes (for example when developer marked it with "aligned" attribute) and pmemobj will not respect that.

Reply:
While very unlikely, I've changed it so that the user has to give us a 'ptr'.